semantic_router/routers/hybrid.py

```diff
@@ -19,13 +19,31 @@
 from semantic_router.utils.logger import logger
 
 
+def fit_sparse_encoder(func):
+    """Decorator that fits the sparse encoder after the decorated function runs.
+    Only fits if the encoder is TfidfEncoder or BM25Encoder and has a fit method."""
+
+    def wrapper(self, *args, **kwargs):
+        result = func(self, *args, **kwargs)
+        if (
+            isinstance(self.sparse_encoder, (TfidfEncoder, BM25Encoder))
+            and hasattr(self.sparse_encoder, "fit")
```

Comment:
Is there any need for `isinstance(self.sparse_encoder, (TfidfEncoder, BM25Encoder))`? Would `hasattr(self.sparse_encoder, "fit")` not be enough?

If someone adds a new sparse encoder that requires fitting I could see them missing this